pkg/vpcmodel/externalNetwork.go

```diff
@@ -43,6 +68,27 @@ func getPublicInternetAddressList() []string {
 	}
 }
 
+func (n *NetworkAddressLists) GetPublicInternetIPblocksList() (ipList []*netset.IPBlock,
+	allPublicInternetRanges *netset.IPBlock, err error) {
+	allPublicInternetRanges = netset.NewIPBlock()
+	if len(n.publicInternetAddressList) == 0 {
+		return ipStringsToIPblocks(GetDefaultPublicInternetAddressList())
+	}
+	for _, ipRange := range n.publicInternetAddressList {
+		allPublicInternetRanges = allPublicInternetRanges.Union(ipRange)
+	}
```

Comment:
Let's compute this union only once in `InitNetworkAddressLists` and store the result in `NetworkAddressLists`